### PR TITLE
Add "static" toggle to Interactive Graph widget

### PR DESCRIPTION
**Pull request by benchristel**
## Summary:
Some of our widgets (e.g. Radio) have a "static" mode which can be toggled on
in the exercise editor. When a widget is in static mode, the "answer" to the
widget is displayed, the widget is not graded, and the learner cannot alter
the widget's state.

Content creators need to be able to set interactive graphs to static mode so
that graphs can be added to hints, to demonstrate the solution steps for an
exercise. Previously, we implemented a special "hint mode" for interactive
graphs to accommodate this use case. Hint mode was based on a couple faulty
assumptions:

- We did not clearly understand what static mode was or when it was turned on.
  We confused static mode with the "read only" state of a widget, which occurs
  when a learner has successfully answered the question. Similar to static mode,
  the read-only state makes the widget non-interactable. However, while static
  mode is configurable by content creators, read-only state is not; it is set
  or unset dynamically, based on the learner's progress through an exercise.
- We assumed that graphs in hints would always be non-interactable, so we
  turned on hint mode automatically in hints (and turned it off in question
  stems) rather than giving content creators a manual toggle. This assumption
  turned out to be incorrect; we now know of at least one question where a hint
  contains a graph that the learner is supposed to be able to manipulate. See:
  https://khanacademy.slack.com/archives/C067UM1QAR4/p1723045477082029

Hint mode (which this feature replaces) was never released to content creators,
so we are okay to drop support for it.

This PR removes hint mode and adds a "static" toggle to the Interactive
Graph widget editor. It builds on work that @nicolecomputer did in
https://github.com/Khan/perseus/pull/1457. Her feature branch is merged into
this one.

Issue: LEMS-2251, LEMS-2053, LEMS-2054

## Test plan:

- Deploy this to a ZND
- Create an exercise in the Test Everything course
- Create a static interactive graph
- Verify that the preview is non-interactable (with mouse and keyboard) when
  the "static" toggle is on, and interactable when "static" is off.
- Save and publish the exercise.
- Go through the exercise as a learner and verify that you cannot manipulate
  the static graphs.
- Verify that the presence of a static graph does not affect how a question is
  graded.

### CODE REVIEW

**Comment by Myranae** [comment]
Haven't finished properly reviewing, but I believe some of this code is code I also changed in my [bug fix PR](https://github.com/Khan/perseus/pull/1487). It's been approved, but I wanted to manually test before landing. Hoping to do that today, just ran into some local dev server / ZND issues. I'll totally update my PR if this one moves forward first though :)

**Comment by benchristel** [comment]
@Myranae I'll wait for you to land your PR. I need to test this on a ZND before landing it anyway.

**Comment by Myranae** [comment]
@benchristel It's landed :)

File: packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx
Comment (jeremywiebe) [inline packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx:48]: I'm not sure if types from other uses of `static` force this, but can we make this `static: boolean | undefined`?  I think there are many places that use this `X | null | undefined` pattern, but in reality, we rarely distinguish between `null` and `undefined`. 

This is a nitpicky thing, but it's another place where the type is broader/more complicated for no reason.

File: packages/perseus/src/widgets/interactive-graphs/reducer/use-graph-config.ts
Comment (jeremywiebe) [inline packages/perseus/src/widgets/interactive-graphs/reducer/use-graph-config.ts:22]: Guess we'll want to avoid having a Perseus concept named "const" then, eh? 😂

**Review by jeremywiebe — APPROVED** [review]
Cool.  I was initially surprised by the size of the PR (number of files touched) but most of that is removing `hintMode`.  I'm pretty happy we can use `static` mode and not introduce another new concept (assuming it all works out). Thanks Ben!

**Comment by Myranae** [comment]
Doing some manual testing on this, I noticed the hints-renderer stories related to interactive graph hint mode don't work anymore because hint mode isn't a thing. We may want to delete all the interactive graph stories from the Hints Renderer stories since you now have static graph stories and graphs shouldn't automatically be turning non-interactive when copied to the hints section. [This story](http://localhost:6006/?path=/story/perseus-renderers-hints-renderer--with-all-interactive-graphs) and all the interactive graph ones under it I mean.

**Comment by benchristel** [comment]
@jeremywiebe @Myranae I realized I missed a crucial part of this feature, which is displaying the correct answer instead of the start coords when the graph is in static mode. So I'll have to ask for another review.

I'd really like Nicole's eyes on this when she's well again, because I suspect she's already done the research to figure out how displaying the correct answer is supposed to work. I've got a solution, but I'm not sure it follows established patterns.

File: packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx
Comment (benchristel) [reply packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx:49]: The type `boolean | null | undefined` comes from [WidgetProps](https://github.com/Khan/perseus/blob/81db6a703677a3447da04ca05ee1707b7f97a851/packages/perseus/src/types.ts#L582), so I had to use the same type here.

I don't really feel comfortable changing WidgetProps, since it's so widely used (and it's one of our few defenses against writing widgets that can't handle production data).

File: packages/perseus/src/__tests__/renderer.test.tsx
Comment (jeremywiebe) [inline packages/perseus/src/__tests__/renderer.test.tsx:1355]: Really glad we don't need to further commit to this hack. 😅

File: packages/perseus/src/widgets/__testdata__/interactive-graph.testdata.ts
Comment (jeremywiebe) [inline packages/perseus/src/widgets/__testdata__/interactive-graph.testdata.ts:853]: Hrm... seems like we now we need a `radioQuestionBuilder()` 😃

File: packages/perseus/src/widgets/interactive-graphs/stateful-mafs-graph.tsx
Comment (jeremywiebe) [inline packages/perseus/src/widgets/interactive-graphs/stateful-mafs-graph.tsx:132]: This becomes obvious fairly quickly as I read the code, but could you leave a comment here that if we're in `static` mode we display the correct answer? (unless you don't feel that's necessary)

**Review by jeremywiebe — APPROVED** [review]
This is great. LGTM!

**Pull request merged**
